mycore-restapi/src/main/java/org/mycore/restapi/v2/MCRRestAuthorizationFilter.java

```diff
@@ -138,9 +141,26 @@ public void filter(ContainerRequestContext requestContext) {
         if (HttpMethod.OPTIONS.equals(method)) {
             return;
         }
-        final String permission
-            = Optional.ofNullable(resourceInfo.getResourceMethod().getAnnotation(MCRRestRequiredPermission.class))
-                .map(MCRRestRequiredPermission::value).orElseGet(() -> getPermissionFromHttpMethod(method));
+        final String permission = Optional
+            .ofNullable(resourceInfo.getResourceMethod().getAnnotation(MCRRestRequiredPermission.class)).map(a -> {
+                final String requiredPermission = a.value();
+                final Class<? extends MCRRestRequiredPermissionResolver> requiredPermissionResolver = a.resolver();
+                if (requiredPermission.isEmpty()
+                    && MCRRestRequiredPermissionDefaultResolver.class.equals(requiredPermissionResolver)) {
+                    LogManager.getLogger("explicit value and resolver is not allowed");
+                    throw new InternalServerErrorException();
+                } else if (!requiredPermission.isEmpty()) {
+                    return requiredPermission;
+                } else {
+                    try {
+                        return requiredPermissionResolver.getDeclaredConstructor().newInstance()
+                            .resolvePermission(requestContext);
+                    } catch (Exception e) {
+                        LogManager.getLogger().error(e);
+                        throw new InternalServerErrorException();
```

Comment:
Should be `new InternalServerErrorException(e);` so the original exception won't get lost where it is eventually logged/evaluated.

Reply:
159 `LogManager.getLogger().error(e);` should be removed since the exception is thrown